### ISSUE TITLE
`systemProperties` contains absolute paths, creating remote build cache misses

### ISSUE THREAD

**Issue by martinbonnin** [JetBrains/gradle-intellij-plugin#1376]
Hi 👋 

Trying to enable remote build cache on apollo-kotlin, `:intellij-plugin:test` creates a cache misses ([build scan](https://ge.apollographql.com/c/akstifgsdtzli/4dnomw3hfzivq/task-inputs?cacheability=cacheable,overlapping-outputs,validation-failure#6elfqwtwxwvus)).

This is due to the `systemProperties` input containing absolute paths. For an example, dumping the value of `systemProperties`:

```kotlin
tasks.named("test") {
  doFirst {
    println(inputs.properties.get("systemProperties"))
  }
}
```

Dumps this:

```
{idea.config.path=/Users/mbonnin/git/apollo-kotlin/intellij-plugin/build/idea-sandbox/config-test,
 idea.force.use.core.classloader=true, idea.home.path=/Users/mbonnin/git/apollo-kotlin/intellij-plugin/mockJDK, 
idea.log.path=/Users/mbonnin/git/apollo-kotlin/intellij-plugin/build/idea-sandbox/system-test/log, 
idea.plugins.path=/Users/mbonnin/git/apollo-kotlin/intellij-plugin/build/idea-sandbox/plugins-test, 
idea.system.path=/Users/mbonnin/git/apollo-kotlin/intellij-plugin/build/idea-sandbox/system-test, idea.use.core.classloader.for=,
 idea.use.core.classloader.for.plugin.path=true, java.system.class.loader=com.intellij.util.lang.PathClassLoader, 
jdk.attach.allowAttachSelf=true, jdk.http.auth.tunneling.disabledSchemes="", jdk.module.illegalAccess.silent=true, 
kotlinx.coroutines.debug=off, plugin.path=/Users/mbonnin/git/apollo-kotlin/intellij-plugin/build/idea-sandbox/plugins-
test/toml:,/Users/mbonnin/git/apollo-kotlin/intellij-plugin/build/idea-sandbox/plugins-test/js-graphql-intellij-
plugin:,/Users/mbonnin/git/apollo-kotlin/intellij-plugin/build/idea-sandbox/plugins-test/apollo-intellij-plugin, 
sun.io.useCanonCaches=false, sun.java2d.metal=true}
```

Would it be possible to make those relative instead?

**Comment by runningcode** [comment]
Here is how you can change these paths to use relative path sensitivity: https://docs.gradle.org/current/userguide/caching_java_projects.html#dealing_with_file_paths

**Comment by hsz** [comment]
Could you folks please verify the fix with `1.14.2-SNAPSHOT` release?
